client/client-lib/src/lib.rs

```diff
@@ -243,6 +243,21 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
         Ok(OutPoint { txid, out_idx: 0 })
     }
 
+    /// Validate tokens without claiming them. This function checks if signatures are valid
+    /// based on the federation public key. It does not check if the nonce is unspent.
+    pub async fn validate_tokens(&self, coins: Coins<SpendableCoin>) -> Result<()> {
+        let tbs_pks = &self.mint_client().context.config.tbs_pks;
+        if coins
+            .iter()
+            .map(|(amt, coin)| coin.coin.verify(*tbs_pks.tier(&amt).unwrap()))
```

Comment:
I tried to use `?` instead of `unwrap` here but couldn't get it to compile. I was trying to do something similar to this https://github.com/fedimint/fedimint/blob/85e6602d6b3df79072bc999b1f966ef90d89d4f2/client/client-lib/src/mint/mod.rs#L262

Reply:
I added a variant to `ClientError` and replaced the `unwrap` with a `?`